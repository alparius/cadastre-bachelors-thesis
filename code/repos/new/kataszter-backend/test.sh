#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
