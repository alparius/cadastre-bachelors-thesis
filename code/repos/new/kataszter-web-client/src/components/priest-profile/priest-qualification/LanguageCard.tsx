import React from "react";
import { FormattedMessage as FM } from "react-intl";
import { Card, Form, Header, Input, List } from "semantic-ui-react";

import ILanguageSkills from "../../../data/priest/LanguageSkills";
import "../../../static/priestProfile.css";
import BoldFM from "../../../util/styledMessages/BoldFM";
import ItalicFM from "../../../util/styledMessages/ItalicFM";

interface ILanguageCardProps {
    languageSkills: ILanguageSkills;
    editMode: boolean;
    label: string;
    handleChangeOnLanguageSkills: (labelName: string, value: any) => void;
}

const LanguageCard: React.FC<ILanguageCardProps> = (props: ILanguageCardProps) => {
    const { languageSkills, editMode, handleChangeOnLanguageSkills, label } = props;

    const handleChange = (_: any, { name, value }: any) => {
        handleChangeOnLanguageSkills(label, { ...languageSkills, [name]: value });
    };
    return (
        <Card fluid>
            <Card.Content>
                <Card.Content>
                    <Card.Header id="header">
                        <Header as="h2">
                            <FM id={"profile." + label} />
                        </Header>
                    </Card.Header>
                </Card.Content>

                <Card.Description>
                    {!languageSkills.well && !languageSkills.medium && !languageSkills.less && !editMode && <ItalicFM id="general.noData" />}
                    <List>
                        {(languageSkills.well || editMode) && (
                            <List.Item>
                                <Form>
                                    <Form.Field inline>
                                        <BoldFM id={"profile.well"} showColon />{" "}
                                        {editMode ? (
                                            <Input size="small" value={languageSkills.well} name="well" onChange={handleChange} fluid />
                                        ) : (
                                            languageSkills.well
                                        )}
                                    </Form.Field>
                                </Form>
                            </List.Item>
                        )}
                        {(languageSkills.medium || editMode) && (
                            <List.Item>
                                <Form>
                                    <Form.Field inline>
                                        <BoldFM id={"profile.medium"} showColon />{" "}
                                        {editMode ? (
                                            <Input size="small" value={languageSkills.medium} name="medium" onChange={handleChange} fluid />
                                        ) : (
                                            languageSkills.medium
                                        )}
                                    </Form.Field>
                                </Form>
                            </List.Item>
                        )}
                        {(languageSkills.less || editMode) && (
                            <List.Item>
                                <Form>
                                    <Form.Field inline>
                                        <BoldFM id={"profile.less"} showColon />{" "}
                                        {editMode ? (
                                            <Input size="small" value={languageSkills.less} name="less" onChange={handleChange} fluid />
                                        ) : (
                                            languageSkills.less
                                        )}
                                    </Form.Field>
                                </Form>
                            </List.Item>
                        )}
                    </List>
                </Card.Description>
            </Card.Content>
        </Card>
    );
};

export default LanguageCard;
