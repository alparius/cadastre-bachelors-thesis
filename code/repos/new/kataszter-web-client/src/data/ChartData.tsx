interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
