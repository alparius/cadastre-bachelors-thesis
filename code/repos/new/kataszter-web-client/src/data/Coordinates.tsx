interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
