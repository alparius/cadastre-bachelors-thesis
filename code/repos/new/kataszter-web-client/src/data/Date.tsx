interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
