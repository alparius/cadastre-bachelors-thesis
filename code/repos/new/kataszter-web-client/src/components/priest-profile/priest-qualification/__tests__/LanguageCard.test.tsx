/*===================
  LanguageCard.unit.test.tsx
 ===================*/
import React from "react";
import ILanguageSkills, { NewLanguageSkills } from "../../../../data/priest/LanguageSkills";
import { NewTestWritingSkills } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntl } from "../../../../util/test/test-wrapper";
import LanguageCard from "../LanguageCard";

const handlers = {
    handleChangeOnLanguageSkills: jest.fn()
};

describe("PriestCard element", () => {
    afterEach(() => {
        handlers.handleChangeOnLanguageSkills.mockClear();
    });
    it("should render a card with the language data", () => {
        const languageSkills: ILanguageSkills = NewTestWritingSkills;
        const container = createComponentWithIntlAndRouter(
            <LanguageCard languageSkills={languageSkills} editMode label={"writingSkills"} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a card with noData label when the language is empty", () => {
        const emptyLanguageSkills: ILanguageSkills = NewLanguageSkills;
        const container = createComponentWithIntlAndRouter(
            <LanguageCard languageSkills={emptyLanguageSkills} editMode label={"writingSkills"} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it.each([
        [0, "testText", "textTest"],
        [1, "testText", "textTest"],
        [2, "testText", "textTest"]
    ])(
        "should  call handleChangeOnLanguageSkills on %d. input. After first change: %s, After second change: %s",
        (inputNr, firstChange, secondChange) => {
            const languageSkills: ILanguageSkills = NewTestWritingSkills;
            const container = mountWithIntl(<LanguageCard languageSkills={languageSkills} editMode label={"writingSkills"} {...handlers} />);
            const input = container.find("input").at(Number(inputNr));
            input.simulate("change", { target: { value: firstChange } });
            input.simulate("change", { target: { value: secondChange } });
            expect(handlers.handleChangeOnLanguageSkills).toHaveBeenCalledTimes(2);
        }
    );
});
