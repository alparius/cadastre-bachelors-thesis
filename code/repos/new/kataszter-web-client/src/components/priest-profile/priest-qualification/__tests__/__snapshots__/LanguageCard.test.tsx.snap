// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestCard element should render a card with noData label when the language is empty 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="content"
    >
      <div
        className="header"
        id="header"
      >
        <h2
          className="ui header"
        >
          Writing skills
        </h2>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Well
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  name="well"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Medium
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  name="medium"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Less
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  name="less"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PriestCard element should render a card with the language data 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="content"
    >
      <div
        className="header"
        id="header"
      >
        <h2
          className="ui header"
        >
          Writing skills
        </h2>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Well
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  name="well"
                  onChange={[Function]}
                  type="text"
                  value="magyar"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Medium
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  name="medium"
                  onChange={[Function]}
                  type="text"
                  value="román"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Less
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  name="less"
                  onChange={[Function]}
                  type="text"
                  value="német"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
